tests/pipe/main.c

```diff
@@ -49,9 +49,10 @@ static void *run_middle(void *arg)
     while (read_total < BYTES_TOTAL) {
         char buf[4];
         unsigned read = pipe_read(&pipes[0], buf, sizeof (buf));
+        buf[read] = 0;
```

Comment:
Potential buffer overflow by one byte. (`pipe_read()` reads up to `sizeof(buf)` bytes. I suggest to increase the buffer by one byte and read only `sizeof(buf) - 1` bytes. Same below.

Reply:
Makes sense, thanks!

Reply:
And done!